### PR TITLE
Add `end_session_endpoint` support

### PR DESCRIPTION
**Pull request by stormsilver**
https://openid.net/specs/openid-connect-session-1_0.html#OPMetadata

### CODE REVIEW

File: lib/openid_connect/client.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module OpenIDConnect
   class Client < Rack::OAuth2::Client
-    attr_optional :userinfo_endpoint, :expires_in
+    attr_optional :userinfo_endpoint, :end_session_endpoint, :expires_in
```
Comment (nov) [inline lib/openid_connect/client.rb]: I don't think you need `end_session_endpoint` and `end_session_uri` in this class.
What is your use-case for them?

**Comment by stormsilver** [comment]
Hi @nov!

The use case is for initiating logout in the OP from a relying party. This does require the RP to redirect to the `end_session_uri` at the end of its own logout procedures.

> URL at the OP to which an RP can perform a redirect to request that the End-User be logged out at the OP.

This change simply allows the `OpenIDConnect::Discovery::Provider::Config::Response` to contain `end_session_url` so that it can be discovered by RPs.

**Comment by nov** [comment]
Sorry for my late response.
Why your commit includes `Client#end_session_uri` then?
I think adding `end_session_uri` to `OpenIDConnect::Discovery::Provider::Config::Response` should be fine.

**Comment by theirishpenguin** [comment]
I would also like to see end_session_endpoint support added.

**Comment by theirishpenguin** [comment]
Thanks @nov @stormsilver 👍

**Pull request closed without merging**